sdks/js/packages/core/react/components/organization/billing/cycle-switch/index.tsx

```diff
@@ -0,0 +1,185 @@
+import { Dialog, Flex, Text, Image, Separator, Button } from '@raystack/apsara';
+import Skeleton from 'react-loading-skeleton';
+import { useNavigate, useParams } from '@tanstack/react-router';
+import cross from '~/react/assets/cross.svg';
+import styles from '../../organization.module.css';
+import { useCallback, useEffect, useState } from 'react';
+import { useFrontier } from '~/react/contexts/FrontierContext';
+import { V1Beta1Plan } from '~/src';
+import { toast } from 'sonner';
+import { getPlanIntervalName, getPlanPrice } from '~/react/utils';
+import * as _ from 'lodash';
+import { usePlans } from '../../plans/hooks/usePlans';
+import dayjs from 'dayjs';
+import { DEFAULT_DATE_FORMAT } from '~/react/utils/constants';
+
+export function ConfirmCycleSwitch() {
+  const { activePlan, client, paymentMethod, config } = useFrontier();
+  const navigate = useNavigate({ from: '/billing/cycle-switch/$planId' });
+  const { planId } = useParams({ from: '/billing/cycle-switch/$planId' });
+  const dateFormat = config?.dateFormat || DEFAULT_DATE_FORMAT;
+
+  const [isPlanLoading, setIsPlanLoading] = useState(false);
+  const [nextPlan, setNextPlan] = useState<V1Beta1Plan>();
+  const [isCycleSwitching, setCycleSwitching] = useState(false);
+
+  const closeModal = useCallback(
+    () => navigate({ to: '/billing' }),
+    [navigate]
+  );
+
+  const {
+    checkoutPlan,
+    isLoading: isPlanActionLoading,
+    changePlan,
+    verifyPlanChange
+  } = usePlans();
+
+  const nextPlanPrice = nextPlan ? getPlanPrice(nextPlan) : { amount: 0 };
+  const isPaymentRequired =
+    _.isEmpty(paymentMethod) && nextPlanPrice.amount > 0;
+
+  const nextPlanIntervalName = getPlanIntervalName(nextPlan);
+
+  const nextPlanMetadata = nextPlan?.metadata as Record<string, number>;
+  const activePlanMetadata = activePlan?.metadata as Record<string, number>;
+
+  const isUpgrade =
+    (Number(nextPlanMetadata?.weightage) || 0) -
+      (Number(activePlanMetadata?.weightage) || 0) >
+    0;
+
+  useEffect(() => {
+    async function getNextPlan(nextPlanId: string) {
+      setIsPlanLoading(true);
+      try {
+        const resp = await client?.frontierServiceGetPlan(nextPlanId);
+        const plan = resp?.data?.plan;
+        setNextPlan(plan);
+      } catch (err: any) {
+        toast.error('Something went wrong', {
+          description: err.message
+        });
+        console.error(err);
+      } finally {
+        setIsPlanLoading(false);
+      }
+    }
+    if (planId) {
+      getNextPlan(planId);
+    }
+  }, [client, planId]);
+
+  const isLoading = isPlanLoading;
+
+  async function onConfirm() {
+    setCycleSwitching(true);
+    try {
+      if (nextPlan?.id) {
```

Comment:
In what scenario will we not have `nextPlan.id`? This method is on clicking "confirm" of the dialogue box, right?

Reply:
The schema type generated by (swagger-typescript-api)[https://github.com/acacode/swagger-typescript-api] has all keys as optional.
I will fix that in separate PR

Reply:
Okay